Src/Support/Google.Apis.Auth/OAuth2/Responses/TokenResponse.cs

```diff
@@ -106,18 +106,25 @@ public bool IsExpired(IClock clock)
             return IssuedUtc.AddSeconds(ExpiresInSeconds.Value - TokenRefreshTimeWindowSeconds) <= clock.UtcNow;
         }
 
+        // Unfortunaly the method above is public so we can't change it.
+        // This method is to be used by the TokenRefreshManager to determine if a token
+        // needs refreshing.
+        // It should pass checkAccessToken:true when refreshing access_token so as to keep the
+        // semantics of the method above.
+        // It should pass checkAccessToken:false when refreshing id_token because in that case
```

Comment:
Why wouldn't we be refreshing everything? It feels odd for a TokenResponse to be sort of "half valid".